### PR TITLE
SideSetsAroundSubdomain change

### PR DESCRIPTION
**Pull request by WilkAndy**
Refs #9447 

Since my MOOSE takes quite a few hours to compile i'm just using the TestHarness to see if this fixes the problem.

### CODE REVIEW

**Comment by WilkAndy** [comment]
I'm sure someone will see a better way of fixing this bug, but removing the BlockRestrictable seems to work at least, and identifies the source of the error.

File: framework/src/meshmodifiers/SideSetsAroundSubdomain.C
Comment (permcody) [inline framework/src/meshmodifiers/SideSetsAroundSubdomain.C]: This PR looks ok except you may as well keep the warning message that is already there rather than just silently ignore all of the boundaries past the first one. I'm not entirely sure why `BlockRestrictable` doesn't work here. Maybe @aeslaughter can take a look but he's currently out and will be busy for the next several weeks. We don't have to hold this up if you need it.

Comment (WilkAndy) [inline framework/src/meshmodifiers/SideSetsAroundSubdomain.C]: Oh, the SubdomainName is now just a single parameter - it's not a vector of Names.  So the warning message is irrelevant.

I don't understand why this MeshModifier originally derived from ```BlockRestrictable```.  No others do, so that's why i was a bit more confident about removing it.  I wonder whether it was just a historical quick-fix way to have a "block = ..." in the input file.  Anyway, clearly ```BlockRestrictable``` doesn't play nicely with adding blocks if you use it naively.

I'd like to merge this PR.  I know what "busy for the next several weeks" means!

Comment (permcody) [inline framework/src/meshmodifiers/SideSetsAroundSubdomain.C]: Oh I see. Well there is a better API that you can use that will clean this up, use the non-vector version:
https://github.com/idaholab/moose/blob/devel/framework/include/mesh/MooseMesh.h#L556

Then you can drop the braces and the brackets. After that I'll merge.

**Review by permcody — CHANGES_REQUESTED** [review]
Use `MooseMesh::getSubdomainID()`

**Comment by WilkAndy** [comment]
Lots of build failures in Material models in Mammoth ( @jasondhales  ?  @bwspenc ?)   (Unrelated to this PR)

**Comment by permcody** [comment]
We'll take care of those failures while you snooze!

File: framework/src/meshmodifiers/SideSetsAroundSubdomain.C
Comment (permcody) [inline framework/src/meshmodifiers/SideSetsAroundSubdomain.C]: Much better

**Review by permcody — APPROVED** [review]

**Pull request merged**
